src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -710,14 +710,20 @@ static void xocl_mailbox_srv(void *arg, void *data, size_t len,
 	struct xcl_mailbox_req *req = (struct xcl_mailbox_req *)data;
 	struct xcl_mailbox_peer_state *st = NULL;
 	struct xclErrorLast err_last;
+	struct xcl_firewall fw_stat = { 0 };
+	XOCL_TIMESPEC time;
 
 	if (err != 0)
 		return;
 
 	userpf_info(xdev, "received request (%d) from peer\n", req->req);
-
 	switch (req->req) {
 	case XCL_MAILBOX_REQ_FIREWALL:
+		xocl_af_get_data(xdev, &fw_stat);
```

Comment:
This will not work. You may expand check_firewall() to request data from remote side and record timestamp for xocl.

Reply:
I updated check_firewall to populate a `struct xcl_firewall`. The idea being that to get cached data call xocl_af_data, and to get the most recent data call xocl_af_check.